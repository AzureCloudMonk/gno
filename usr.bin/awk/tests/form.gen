# form.gen - generate form letters
#   input:  prototype file letter.text; data lines
#   output: one form letter per data line

BEGIN {
    oldfs = FS;
    FS = ":"
    while (getline <"letter.text" > 0) # read form letter
        form[++n] = $0
    FS = oldfs;
    # get country name from command-line argument
    pat = ARGV[1]; ARGV[1] = "-"
}

$1 ~ pat {   for (i = 1; i <= n; i++) { # read data line for country
        temp = form[i]
        for (j = 1; j <= 3; j++)
            gsub("#" j, $j, temp)
	gsub("#4", $2/$3, temp)
        print temp
    }
}
