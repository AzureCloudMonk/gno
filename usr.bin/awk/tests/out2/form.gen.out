Subject: Demographic Information About England
From: AWK Demographics, Inc.

In response to your request for information about England,
our latest research has revealed that its population is 94
million people and its area is 56 million square miles.
This gives England a population density of 1.67857 people per
square mile.
